cmd/gen-go-sample/out.go

```diff
@@ -145,6 +150,14 @@ func writePrint(pFmt string, pArgs []string, st *symTab, gen *generator) error {
 }
 
 func writeLoop(l *LoopSpec, st *symTab, gen *generator) error {
+	if l.Variable == "" {
+		return errors.E(nil, "variable not specified for looping over arrays")
+	}
+
+	if l.Map != "" || l.Key != "" || l.Value != "" {
```

Comment:
Checks like this break encapsulation: why should the loop-specific function check any entries for maps? This is why I suggest the more centralized check above.